icaruscode/CRT/CRTPMTMatchingAna_module.cc

```diff
@@ -174,300 +312,391 @@ icarus::crt::CRTPMTMatchingAna::CRTPMTMatchingAna(fhicl::ParameterSet const& p)
-    fMatchTree->Branch("crtRegion",            &fCrtRegion);
-    fMatchTree->Branch("tofHit",               &fTofHit);
-    fMatchTree->Branch("tofFlash",             &fTofFlash);
-    fMatchTree->Branch("tofFlashHit",          &fTofFlashHit);
-    fMatchTree->Branch("peHit",                &fTofPeHit);
-    fMatchTree->Branch("peFlash",              &fTofPeFlash);
-    fMatchTree->Branch("peFlashHit",           &fTofPeFlashHit);
-    fMatchTree->Branch("xyztHit",              &fTofXYZTHit);
-    fMatchTree->Branch("xyztFlash",            &fTofXYZTFlash);
-    fMatchTree->Branch("xyztFlashHit",         &fTofXYZTFlashHit);
-    fMatchTree->Branch("distHit",              &fDistHit);
-    fMatchTree->Branch("distFlash",            &fDistFlash);
-    fMatchTree->Branch("distFlashHit",         &fDistFlashHit);
-    fMatchTree->Branch("tpcHit",               &fTofTpcHit);
-    fMatchTree->Branch("tpcFlash",             &fTofTpcFlash);
-    fMatchTree->Branch("gate_type",            &m_gate_type, "gate_type/b");
-    fMatchTree->Branch("gate_name",            &m_gate_name);
-    fMatchTree->Branch("trigger_timestamp",    &m_trigger_timestamp, "trigger_timestamp/l");
-    fMatchTree->Branch("gate_start_timestamp", &m_gate_start_timestamp, "gate_start_timestamp/l");
-    fMatchTree->Branch("trigger_gate_diff",    &m_trigger_gate_diff, "trigger_gate_diff/l");
-    fMatchTree->Branch("gate_crt_diff",        &m_gate_crt_diff, "gate_crt_diff/l");
+  // Call appropriate consumes<>() for any products to be retrieved by this module.
+  fFlashLabels[0] = fOpFlashModuleLabel0;
+  fFlashLabels[1] = fOpFlashModuleLabel1;
+  fFlashLabels[2] = fOpFlashModuleLabel2;
+  fFlashLabels[3] = fOpFlashModuleLabel3;
+
+  // Get a pointer to the geometry service provider.
+  fGeometryService = lar::providerFrom<geo::Geometry>();
+
+  art::ServiceHandle<art::TFileService> tfs;
+
+  fMatchTree = tfs->make<TTree>("matchTree","CRTHit - OpHit/Flash matching analysis");
+
+  fMatchTree->Branch("event",        &fEvent,          "event/I");
+  fMatchTree->Branch("run",          &fRun,            "run/I");
+  fMatchTree->Branch("subrun",       &fSubRun,         "subrun/I");
+  fMatchTree->Branch("opFlash_pos",  &fOpFlashPos);
+  fMatchTree->Branch("opFlash_time",  &fOpFlashTime);
+  fMatchTree->Branch("opFlashAbsTime", &fOpFlashTimeAbs);
+  fMatchTree->Branch("opFlash_time_firsthit",  &fOpFlashTimehit);
+  fMatchTree->Branch("opHit_x",   &fOpHitX);
+  fMatchTree->Branch("opHit_y",   &fOpHitY);
+  fMatchTree->Branch("opHit_z",   &fOpHitZ);
+  fMatchTree->Branch("opHit_t",   &fOpHitT);
+  fMatchTree->Branch("opHit_amplitude",   &fOpHitA); 
+  fMatchTree->Branch("inTime",  &fInTime);
+  fMatchTree->Branch("inTime_beam",  &fInTime_beam);
+  fMatchTree->Branch("inTime_gate",  &fInTime_gate);
+  fMatchTree->Branch("CRT_matches",     &fNCRTmatch);
+  fMatchTree->Branch("CRT_pos",  &fMatchedCRTpos);
+  fMatchTree->Branch("CRT_Time",  &fMatchedCRTtime);
```

Comment:
Similarly, maybe it would be clearer to add the units to the branch name, e.g., CRT_Time_us?

Reply:
done!